src/Backend/OnboardingInsuranceAPI/Areas/Insurance/GetContribution.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.Enums;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class GetContribution : IHandler
+{
+    private readonly DataContext _context;
+
+    public GetContribution(DataContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<ContributionData> Handle(ContributionData requestedData, string pid)
+    {
+        if(string.IsNullOrEmpty(pid))
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+
+        var birthDate = _context.Users.FirstOrDefault(u => u.Pid == pid).Birthdate;
```

Comment:
Again and again. FirstOrDefault can and will return Default => null. Here it would throw Null Reference Exception.

Reply:
You could use "?" after it so that it returns null either way, since you are checking for nullorempty right after it.

Reply:
Ok get it, becuase its an object and we are looking in null property .Birthdate => null.Birthdate and its not possible, so I fixed as You adviced.